src/Pagination/Goto/Goto.components.tsx

```diff
@@ -0,0 +1,63 @@
+// VENDOR
+import * as React from 'react';
+import classNames from 'classnames';
+import styled from '@xstyled/styled-components';
+// ANCHOR
+import { Input } from '../..';
+
+// Todo: this component needs work for mobile and for accessibility.
+
+const EventKeyCodes = {
+    ENTER: 13,
+};
+
+interface GotoProps {
+    className?: string;
+    onSubmit: (arg: number) => any;
+}
+
+const StyledGoto = styled('div')`
+    width: 6rem;
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    padding: 0 0.5rem;
+
+    .anchor-input {
+        margin-left: 0.5rem;
+    }
+`;
+
+export const Goto = ({
+    onSubmit,
+    className,
+    ...props
+}: GotoProps): React.ReactElement<GotoProps> => {
+    const [value, setValue] = React.useState<null | number>(null);
```

Comment:
Why not always zero (`0`) this out? Is it because you'd like an active state in case the DOM displayed numbers do not include the active page, or some form of allowing a state without any active element?

Reply:
Not sure exactly what you're asking, but `null | number` is effectively the Type of the `Input type="number"` value. I decided to keep that for the Type of the state here, with `null` representing an empty input.

Reply:
I'm on the fence, between two tensions: (a) this should always have a value even if that value is 0 because you're always on a page in a paginated set or (b) allow the user to clear the value if they want.

Reply:
In the current route I went with, the Goto is empty and doesn't track the current page. If a user deigns to enter a value, then it'll go to that page and clear itself. (If the value the user entered it too small or large, it goes to the first or last page respectively. A value of `0` would be outside the range of viable pages to go to

Reply:
That behavior does make sense and we don't exactly have UX insight on this, so I'll resolve this. Thanks for the context!